### DISCUSSION TITLE
Industry and product contribution analysis

### DISCUSSION THREAD

**Discussion by marpellan** [SESAM-Polimi/MARIO#32] [Ideas]
Dear MARIO team, 

I am just discovering MARIO, it looks great! I was using pymrio before, but it looks like there are lot of useful functions here. 

I wonder if you have implemented functions to perform industry and product contribution analysis, as described in the article of Wiedmann (2017) 'On the decomposition of total impact multipliers in a supply and use framework' (https://ideas.repec.org/a/spr/jecstr/v6y2017i1d10.1186_s40008-017-0072-0.html). I had difficulties implementing the product decomposition with pymrio. 

Thanks a lot, 

Best regards

**Comment by nigolred** [comment]
Dear @marpellan,

Sorry for the late reply, we really missed your much appreciated comment.
Give us a few days to read Wiedmann's article carefully.

Kind regards,

Nicolò

**Reply by nigolred** [reply]
Dear @marpellan,

I apologise again for the long delay.

Anyway, the product decomposition calculation is easily implemented on MARIO. We have been using it since we made MARIO, but Wiedmann's publication that you shared validates this very useful methodology for visualising impacts.

In particular, you only need to import a database (say world = mario.parse_from_excel(...)):
- take the row of satellite accounts that you are interested in. Let's imagine it is the CO2 row.
- diagonalise world.e.loc['CO2'] (let's call it e_diag)
- calculate the product between the footprint of the same environmental extension (world.f.loc['CO2'] according to the MARIO nomenclature, let's call it f_diag) and the technology coefficient matrix (it is called z in MARIO and is also present for the SUT framework by making world.z)

Ultimately, its matrix is e_diag + f_diag @ z.

Take care to align the indices or do the calculations using only the values of these objects (which are DataFrame pandas).

Do you think it would be convenient to have this calculation already present in MARIO as a function? For example, world.PDI or something like that?

PS: if the nomenclature confuses you, know that we are working on a new version of MARIO that allows you to call the matrices as you are used to (since they change from search group to search group)
